### PR TITLE
[PA-662] Modify GetPxBackupNamespace to detect PxBackup namespace

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR fixes an issue in the Px-Backup automation where it was previously limited to working only in the "px-backup" namespace. With this PR, the automation can now be executed in any namespace, making it more flexible and adaptable to different deployment scenarios.

**Which issue(s) this PR fixes** (optional)
Closes #PA-662

**Special notes for your reviewer**:
Please review the Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "px-backup" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/105/

Please review the Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "central" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/108/

Please review the Jenkins build for the "BasicBackupCreation" test case with no px-backup using the link below [FAILED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/107

### CODE REVIEW

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

File: drivers/backup/auth.go [hunk 1]
```diff
@@ -213,12 +229,18 @@ func getKeycloakEndPoint(admin bool) (string, error) {
...
 
 // GetPxBackupNamespace returns namespace of px-backup deployment.
-func GetPxBackupNamespace() string {
-	ns, present := os.LookupEnv(pxBackupNamespace)
-	if !(present) {
-		return AdminTokenSecretNamespace
+func GetPxBackupNamespace() (string, error) {
+	var allServices *corev1.ServiceList
+	var err error
+	if allServices, err = k8sCore.ListServices("", metav1.ListOptions{}); err != nil {
```
Comment (nikolaypopov) [inline drivers/backup/auth.go:235]: nit: just remove defining vars for `err` and `allServices` and do
```
allServices, err := k8sCore.ListServices("", metav1.ListOptions{})
if err != nil {
    return "", fmt.Errorf("failed to get list of services. Err: %v", err)
}
```

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -6308,7 +6308,9 @@ var _ = Describe("{RestartBackupPodDuringBackupSharing}", func() {
 			log.InfoD("Restart backup pod when backup sharing is in-progress")
 			backupPodLabel := make(map[string]string)
 			backupPodLabel["app"] = "px-backup"
-			err := DeletePodWithLabelInNamespace(backup.GetPxBackupNamespace(), backupPodLabel)
+			pxbNamespace, err := backup.GetPxBackupNamespace()
+			dash.VerifyFatal(err, nil, "Getting px-backup namespace")
+			err = DeletePodWithLabelInNamespace(pxbNamespace, backupPodLabel)
 			dash.VerifyFatal(err, nil, "Restart backup pod when backup sharing is in-progress")
```
Comment (nikolaypopov) [inline tests/backup/backup_test.go:6314]: The function `DeletePodWithLabelInNamespace()` and error `Restart backup pod when backup sharing is in-progress` are not matching, should we have a different error message here? Like `Failed to delete pods with label someLabel in namespace someNamespace"` or something?

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -6364,7 +6366,9 @@ var _ = Describe("{RestartBackupPodDuringBackupSharing}", func() {
 			log.InfoD("Restart mongo pod when backup sharing is in-progress")
 			backupPodLabel := make(map[string]string)
 			backupPodLabel["app.kubernetes.io/component"] = "pxc-backup-mongodb"
-			err := DeletePodWithLabelInNamespace(backup.GetPxBackupNamespace(), backupPodLabel)
+			pxbNamespace, err := backup.GetPxBackupNamespace()
+			dash.VerifyFatal(err, nil, "Getting px-backup namespace")
+			err = DeletePodWithLabelInNamespace(pxbNamespace, backupPodLabel)
 			dash.VerifyFatal(err, nil, "Restart mongo pod when backup sharing is in-progress")
```
Comment (nikolaypopov) [inline tests/backup/backup_test.go:6372]: Same here

File: tests/backup/backup_test.go (hunk 2 above)
Comment (nikolaypopov) [reply tests/backup/backup_test.go:6314]: Also can we use a `sched-ops` function for this https://github.com/portworx/sched-ops/blob/master/k8s/core/pods.go#L61? We don't need to re-write this function `DeletePodWithLabelInNamespace()`, this method already exists

**Review by nikolaypopov — CHANGES_REQUESTED** [review]
Please see comments

File: tests/backup/backup_test.go (hunk 2 above)
Comment (nikolaypopov) [reply tests/backup/backup_test.go:6314]: We should remove this function and use `sched-ops` function everywhere this func is used

Comment (mkoppal-px) [reply tests/backup/backup_test.go:6314]: @nikolaypopov We are not actually trying to delete the pod here. We want to bounce (restart) the pod. The `sched-ops` function which you suggested has a timeout parameter. I am guessing it waits for the pod to be actually deleted and if it doesn't then it context deadline will exceed and it fails right? But here the deployment spec makes sure the pod comes back up and that is what we expect.

Comment (nikolaypopov) [reply tests/backup/backup_test.go:6314]: Discussed with @mkoppal-px, never mind this comment

File: tests/backup/backup_test.go (hunk 3 above)
Comment (nikolaypopov) [reply tests/backup/backup_test.go:6372]: Discussed with @mkoppal-px, never mind this comment

File: drivers/backup/auth.go (hunk 1 above)
Comment (kphalgun-px) [reply drivers/backup/auth.go]: Done.

**Comment by kphalgun-px** [comment]
Please review the latest Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "px-backup" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/134/

Please review the latest Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "central" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/136/

Please review the latest Jenkins build for the "BasicBackupCreation" test case with no px-backup using the link below [FAILED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/135/

**Comment by kphalgun-px** [comment]
Final test build after merging

Please review the latest Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "px-backup" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/146/

Please review the latest Jenkins build for the "BasicBackupCreation" test case with px-backup namespace "central" using the link below [PASSED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/145/

Please review the latest Jenkins build for the "BasicBackupCreation" test case with no px-backup using the link below [FAILED]

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/144/

**Pull request merged**